### PR TITLE
Get Started Walkthrough for vscode knative

### PR DESCRIPTION
**Pull request by sudhirverma**
Fix: #262

### CODE REVIEW

**Review by lstocchi — COMMENTED** [review]
By a quick glance it looks good. I would change something in the descriptions but i'll leave the english review to Mohit as i suck.
Only 3 notes: 
1) maybe it's better to add an action in the `create new function` step that opens up the wizard to create a func. In IJ i cannot do it as the plugin works with all Jetbrains IDEs (IJ, GoLand, Pycharm,...) and they work differently but in vscode i guess it should be easier. 
2) Is it really necessary to have the add/remove env variable/volumes explained? 
3) in the invoke step i would mention the function need to be run before to invoke it.

**Comment by sudhirverma** [comment]
> 2\. Is it really necessary to have the add/remove env variable/volumes explained?

@mohitsuman should we include the add/remove env variable/volumes in getting started or should I skip it?

File: package.json [hunk 1]
```diff
@@ -635,6 +636,96 @@
...
+            "description": "Run your function to see how it works locally before to push it on your cluster.\nOnly a function which has the source code opened in the IDE can be run. Right click on the function you want to run (look for its node in the Functions tree), open the context menu (right-click on the node) and click on \"Run\". The output channel will show up with run command where you can see the logs.",
+            "media": {
+              "image": "https://raw.githubusercontent.com/wiki/redhat-developer/vscode-knative/walkthrough/run.gif",
+              "altText": "run function"
+            }
+          },
+          {
+            "id": "Deploy Function",
+            "title": "Deploy function",
+            "description": "Deploy your function to your cluster directly from your IDE.\nOnly a function which has the source code opened in the IDE can be pushed. Right click on the function you want to deploy (look for its node in the Functions tree), open the context menu (right-click on the node) and click on \"Deploy\". The output channel will show up with deploy command where you can see the logs.",
```
Comment (mohitsuman) [inline package.json:675]:
```suggestion
            "description": "Deploys a function to the currently configured Knative-enabled cluster from your IDE.\nOnly a function which has the source code opened in the IDE can be pushed. Right-click on the function you want to deploy (look for its node in the Functions tree), open the context menu (right-click on the node) and click on \"Deploy\". The Output Channel will show up with the deploy command where you can see the logs.",
```

File: package.json [hunk 2]
```diff
@@ -635,6 +636,96 @@
...
+            "title": "Run the function locally",
+            "description": "Run your function to see how it works locally before to push it on your cluster.\nOnly a function which has the source code opened in the IDE can be run. Right click on the function you want to run (look for its node in the Functions tree), open the context menu (right-click on the node) and click on \"Run\". The output channel will show up with run command where you can see the logs.",
+            "media": {
+              "image": "https://raw.githubusercontent.com/wiki/redhat-developer/vscode-knative/walkthrough/run.gif",
+              "altText": "run function"
+            }
+          },
+          {
+            "id": "Deploy Function",
+            "title": "Deploy function",
```
Comment (mohitsuman) [inline package.json:674]:
```suggestion
            "title": "Deploy the Function",
```

File: package.json [hunk 3]
```diff
@@ -635,6 +636,96 @@
...
+            "description": "Deploy your function to your cluster directly from your IDE.\nOnly a function which has the source code opened in the IDE can be pushed. Right click on the function you want to deploy (look for its node in the Functions tree), open the context menu (right-click on the node) and click on \"Deploy\". The output channel will show up with deploy command where you can see the logs.",
+            "media": {
+              "image": "https://raw.githubusercontent.com/wiki/redhat-developer/vscode-knative/walkthrough/deploy.gif",
+              "altText": "deploy function"
+            }
+          },
+          {
+            "id": "Invoke Function",
+            "title": "Invoke a Function",
+            "description": "Invokes the Function by sending a request to the current running Function instance, either locally or remote.\nOnly a function which has the source code opened in the IDE can be invoked. Right click on the function you want to invoke (look for its node in the Functions tree), open the context menu (right-click on the node) and click on \"Invoke\". Provide all necessary details and click on invoke.",
```
Comment (mohitsuman) [inline package.json:684]:
```suggestion
            "description": "Invokes the function by sending a test request to the currently running function instance, either locally or remote.\nOnly a function which has the source code opened in the IDE can be invoked. Right-click on the function you want to invoke (look for its node in the Functions tree), open the context menu (right-click on the node) and click on \"Invoke\". Provide all necessary details and click on Invoke.",
```

File: package.json [hunk 4]
```diff
@@ -635,6 +636,96 @@
...
+            "media": {
+              "image": "https://raw.githubusercontent.com/wiki/redhat-developer/vscode-knative/walkthrough/invoke.gif",
+              "altText": "invoke function"
+            }
+          },
+          {
+            "id": "Add environment variable",
+            "title": "Add environment variable to the function configuration",
+            "description": "Only a function which has the source code opened in the IDE can be configure to add environment variables.\n Right click on the function you want to add environment variables (look for its node in the Functions tree), open the context menu (right-click on the node) and click on \"Add Config\" and then click on \"Add Environment Variables\". An Interactive prompt will open in terminal to add Environment variables.",
```
Comment (mohitsuman) [inline package.json:693]:
```suggestion
            "description": "Only a function which has the source code opened in the IDE can be configured to add environment variables.\n Right-click on the function you want to add environment variables (look for its node in the Functions tree), open the context menu (right-click on the node) and click on \" Add Config\" and then click on \" Add Environment Variables\". An Interactive prompt will open in the terminal to add Environment variables.",
```

File: package.json [hunk 5]
```diff
@@ -635,6 +636,96 @@
...
+            "media": {
+              "image": "https://raw.githubusercontent.com/wiki/redhat-developer/vscode-knative/walkthrough/envs-add.gif",
+              "altText": "invoke function"
+            }
+          },
+          {
+            "id": "Add volumes",
+            "title": "Add volume to the function configuration.",
+            "description": "Only a function which has the source code opened in the IDE can be configure to add volumes.\n Right click on the function you want to add volume (look for its node in the Functions tree), open the context menu (right-click on the node) and click on \"Add Config\" and then click on \"Add volumes\". An Interactive prompt will open in terminal to add Secrets and ConfigMaps as Volume.",
```
Comment (mohitsuman) [inline package.json:702]:
```suggestion
            "description": "Only a function which has the source code opened in the IDE can be configured to add volumes.\n Right-click on the function you want to add volume (look for its node in the Functions tree), open the context menu (right-click on the node) and click on \" Add Config\" and then click on \" Add volumes\". An Interactive prompt will open in the terminal to add Secrets and ConfigMaps as Volume.",
```

File: package.json [hunk 6]
```diff
@@ -635,6 +636,96 @@
...
+            "media": {
+              "image": "https://raw.githubusercontent.com/wiki/redhat-developer/vscode-knative/walkthrough/add-volume.gif",
+              "altText": "invoke function"
+            }
+          },
+          {
+            "id": "Remove environment variable",
+            "title": "Remove environment variable from the function configuration",
+            "description": "Only a function which has the source code opened in the IDE can be configure to remove environment variables.\n Right click on the function you want to remove environment variables (look for its node in the Functions tree), open the context menu (right-click on the node) and click on \"Remove Config\" and then click on \"Remove Environment Variables\". An Interactive prompt will open in terminal to remove environment variables.",
```
Comment (mohitsuman) [inline package.json:711]:
```suggestion
            "description": "Only a function which has the source code opened in the IDE can be configured to remove environment variables.\n Right-click on the function you want to remove environment variables (look for its node in the Functions tree), open the context menu (right-click on the node) and click on \"Remove Config\" and then click on \"Remove Environment Variables\". An Interactive prompt will open in the terminal to remove environment variables.",
```

File: package.json [hunk 7]
```diff
@@ -635,6 +636,96 @@
...
+            "media": {
+              "image": "https://raw.githubusercontent.com/wiki/redhat-developer/vscode-knative/walkthrough/envs-remove.gif",
+              "altText": "invoke function"
+            }
+          },
+          {
+            "id": "Remove volumes",
+            "title": "Remove volume from the function configuration",
+            "description": "Only a function which has the source code opened in the IDE can be configure to remove volumes.\n Right click on the function you want to remove volume (look for its node in the Functions tree), open the context menu (right-click on the node) and click on \"Remove Config\" and then click on \"Remove volumes\". An Interactive prompt will open in terminal to remove volume mounts from the function.",
```
Comment (mohitsuman) [inline package.json:720]:
```suggestion
            "description": "Only a function which has the source code opened in the IDE can be configured to remove volumes.\n Right-click on the function you want to remove volume (look for its node in the Functions tree), open the context menu (right-click on the node) and click on \"Remove Config\" and then click on \"Remove volumes\". An Interactive prompt will open in the terminal to remove volume mounts from the function.",
```

**Review by mohitsuman — CHANGES_REQUESTED** [review]

**Review by mohitsuman — APPROVED** [review]

**Pull request merged**